phoenix-core/src/main/java/org/apache/phoenix/coprocessor/PhoenixAccessController.java

```diff
@@ -229,17 +227,12 @@ public void handleRequireAccessOnDependentTable(String request, String userName,
                     + dependentTable);
             return;
         }
-        if (isAutomaticGrantEnabled) {
```

Comment:
Does it make sense to also remove the isStrictMode option? We should always keep the index and data table permissions in sync.

Reply:
What is the isStrictMode option?

Reply:
In handleRequireAccessOnDependentTable if the isStrictMode boolean is set to false then the automatic grant code path is skipped.

Reply:
I think , we can keep this option. as it will help admin if he doesn't want to see the permissions problems on the dependent table and wants to manage the permission by himself once the table is created(with downtime) and don't want to rely on the automatic grant.

Reply:
I didn't quite clearly understand the difference between automatic grant and strict mode. Could you please elaborate @ankitsinghal?

Reply:
This patch removes the automatic grant option and now always keep the permissions in sync between the data table and any index tables. You can only execute a GRANT/REVOKE on the data table and not on the index table. So I don't think we need the isStrictMode option any more. Once that is removed the data table and index table permissions for all uses will always be kept in sync.

Reply:
@karanmehta93 
Strict mode:- It will check permissions for dependent tables as well. For eg, If a user who has all access on data table is creating an index, then we need to ensure that all others users of data table can also access a new index table.
AutomaticGrant:- It will automatically grant required permissions to dependent table users.

@twdsilva , what about the case when a new index is been created?
Purpose of the automatic grant:- let's say there are three users A and B have READ permission on the data-table and user C has RWC permission on data-table. so if user B creates an index, then we need to ensure that user A and C should also be able to read the index and C should be able to write to this Index and can drop the index also. so we will give only the required permission to the users of data-table on the index table. So, Access should propagate like this. 

user | access data table | access on index table | with Automatic grant(access on index table will change like this) | comments
-- | -- | -- | -- | --
A | RAX | no access | RX | RX will be given on index table
B | RX | RWXC | RWXC | no grant will happen
C | RWXAC | no access | RWCX | read ,write and create will be given so that it can read/write to index table and drop as well.

Reply:
I don't get the idea of the strict mode option. If this is on, then the grants for secondary indexes would not be done, essentially breaking things. We already have enough config options, so I'd be in favor of removing this one.

Reply:
Oh, I see.. here we are just talking about removing a config, not the automatic grant flow in the coprocessor. I'm fine with either way, having these configs with suitable default or removing the configs completely.

Reply:
> Oh, I see.. here we are just talking about removing a config, not the automatic grant flow in the coprocessor. I'm fine with either having these configs with suitable default or removing the configs completely

Yes the code flow will still remain and the scenario that you have explained is a valid one. I will remove both the configs from the code path everywhere.